File: 18015_18174_ProjetoED/TrensCidades/TrensCidades/Classes/Linha.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;

namespace TrensCidades.Classes
{
    //Gustavo Henrique de Meira - 18015
    //Pedro Gomes Moreira - 18174

    class Linha
    {
        private string conteudo;

        public Linha(
            string conteudo)
        {
            Conteudo = conteudo;
        }

        public string Conteudo { get => conteudo; set => conteudo = value; }
    }
}

File: 18015_18174_ProjetoED/TrensCidadesAndroid/app/src/main/java/com/example/trenscidadesandroid/classes/linha/Linha.java
//Gustavo Henrique de Meira - 18015
//Pedro Gomes Moreira - 18174

package com.example.trenscidadesandroid.classes.linha;

import java.io.Serializable;

//Classe para diferenciar o construtor de nome do construtor em que é passado uma linha inteira
public class Linha
    implements Serializable
{
    //Conteúdo string interno da Linha
    private String conteudo;

    //Construtor com string
    public Linha(
        String conteudo) throws Exception
    {
        setConteudo(conteudo);
    }

    //Getter e setter

    public String getConteudo()
    {
        return conteudo;
    }

    public void setConteudo(
        String conteudo) throws Exception
    {
        if (conteudo == null || conteudo.equals(""))
            throw new Exception("Linha - setConteudo: string inválida");
        this.conteudo = conteudo;
    }
}
